File: pydroidui/src/main/java/com/pyamsoft/pydroid/ui/app/BaseBoundPreference.java
/*
 * Copyright 2016 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.pyamsoft.ui.app;

import android.content.Context;
import android.support.annotation.CallSuper;
import android.support.v7.preference.Preference;
import android.support.v7.preference.PreferenceViewHolder;
import android.util.AttributeSet;

public abstract class BaseBoundPreference extends Preference {

  protected BaseBoundPreference(Context context, AttributeSet attrs, int defStyleAttr,
      int defStyleRes) {
    super(context, attrs, defStyleAttr, defStyleRes);
  }

  protected BaseBoundPreference(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }

  protected BaseBoundPreference(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  protected BaseBoundPreference(Context context) {
    super(context);
  }

  @CallSuper @Override public void onBindViewHolder(PreferenceViewHolder holder) {
    super.onBindViewHolder(holder);
    onUnbindViewHolder();
  }

  @CallSuper @Override public void onDetached() {
    super.onDetached();
    onUnbindViewHolder();
  }

  @SuppressWarnings("WeakerAccess") protected void onUnbindViewHolder() {

  }
}

File: pydroid-ui/src/main/java/com/pyamsoft/pydroid/ui/app/BaseBoundPreference.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.pydroid.ui.app

import android.content.Context
import android.support.annotation.CallSuper
import android.support.v7.preference.Preference
import android.support.v7.preference.PreferenceViewHolder
import android.util.AttributeSet

abstract class BaseBoundPreference : Preference {

  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int,
      defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context,
      attrs, defStyleAttr)

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

  constructor(context: Context) : super(context)

  @CallSuper override fun onBindViewHolder(holder: PreferenceViewHolder) {
    super.onBindViewHolder(holder)
    onUnbindViewHolder(holder)
  }

  @CallSuper override fun onDetached() {
    super.onDetached()
    onUnbindViewHolder(null)
  }

  protected open fun onUnbindViewHolder(holder: PreferenceViewHolder?) {

  }
}
